### PR TITLE
Make bash the default shell on windows

### PR DESCRIPTION
**Pull request by r0qs**
The current matrix notification fails on windows because it should run using bash instead of powershell. See: https://app.circleci.com/pipelines/github/ethereum/solidity/29203/workflows/80133c6d-1c41-49e6-b0be-441ca5cdfe21/jobs/1296903

This PR changes the default shell on the CI to bash and only uses powershell where it is needed.

### CODE REVIEW

**Comment by ekpyron** [comment]
Do you happen to know the reason why powershell is behaving weirdly? I mean, it doesn't ring right to me for solc to just not output anything when invoked via powershell (judging from what I saw on your screen) - or is it rather that it doesn't find the binary and doesn't even execute it without error message?

TL;DR; we should make sure that this isn't actually an issue in solc in some way, i.e. solc should work via powershell.

**Comment by r0qs** [comment]
> Do you happen to know the reason why powershell is behaving weirdly? I mean, it doesn't ring right to me for solc to just not output anything when invoked via powershell (judging from what I saw on your screen) - or is it rather that it doesn't find the binary and doesn't even execute it without error message?

No I couldn't find out why. I think it doesn't even execute it, but it is very strange that it doesn't throw an error.

> TL;DR; we should make sure that this isn't actually an issue in solc in some way, i.e. solc should work via powershell.

Yes, this PR does not change that. The `solc` binary still running using powershell. The PR only changes the default shell on the windows environment to bash, so the matrix notification will work on windows as well without the need to change the current CI logic.

**Comment by ekpyron** [comment]
Well, ok - I didn't mean to suggest that the PR changes this, just that we should be sure that solc does properly work on powershell in general - it's not like it's impossible that this has been broken for whatever reason for ages :-).

**Comment by r0qs** [comment]
> Well, ok - I didn't mean to suggest that the PR changes this, just that we should be sure that solc does properly work on powershell in general - it's not like it's impossible that this has been broken for whatever reason for ages :-).

I just noticed that I was indeed running `solc` over bash instead of powershell for the `lsp` test. I fixed it and also removed a step that I guess is unnecessary, i.e. `Inspect lsp.py`.

**Review by matheusaaguiar — APPROVED** [review]
Can't see anything wrong and tests are green.

**Comment by ekpyron** [comment]
Not sure whether powershell or bash is the saner default here - but we're writing this stuff ourselves and who wants to actively write powershell scripts if we don't need to - so as long as we can still run and test our actual powershell scripts, this is fine and I'm merging.
